indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -86,6 +88,11 @@
  */
 public class TaskLockbox
 {
+  private static final long LOCK_ACQUIRE_TIMEOUT_MILLIS = 10_000L;
+
+  private final Map<String, ReadWriteLock> datasourceToConcurrentLock = new HashMap<>();
```

Comment:
Annotate this `@GuardedBy("giant")`.

Reply:
No longer needed.
I'm using a ConcurrentHasMap now.
The operations are synchronized on the map's value.